File: 0303-range-sum-query-immutable/0303-range-sum-query-immutable.cpp
class NumArray {
public:
    vector<int> res;
    NumArray(vector<int>& nums) {
        res.resize(nums.size());
        res[0] = nums[0];
        res.emplace_back(nums[0]);
        
        for(int i=1;i<nums.size();i++){
            res[i] = res[i-1]+nums[i];
        }
    }
    
    int sumRange(int left, int right) {
        if(left==0)
            return res[right];
        return res[right]-res[left-1];
    }
};

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray* obj = new NumArray(nums);
 * int param_1 = obj->sumRange(left,right);
 */

File: 0303-range-sum-query-immutable/0303-range-sum-query-immutable.py
class NumArray:

    def __init__(self, nums: List[int]):
        self.res= [0 for _ in range(len(nums))]
        self.res[0] = nums[0]
        
        for i in range(1,len(nums)):
            self.res[i] = nums[i]+self.res[i-1]

    def sumRange(self, left: int, right: int) -> int:
        if left==0:
            return self.res[right]
        
        return self.res[right] - self.res[left-1]


# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(left,right)